test/cpp/api/tensor.cpp

```diff
@@ -171,6 +171,13 @@ TEST(TensorTest, ContainsCorrectValuesForManyValues) {
   ASSERT_TRUE(exactly_equal(tensor[1], 2));
   ASSERT_TRUE(exactly_equal(tensor[2], 3));
 
+  tensor = at::tensor(at::ArrayRef<int>({1, 2, 3}));
```

Comment:
I'm confused -- what value does testing with `at::ArrayRef<int>` add?

Reply:
`at::tensor(at::ArrayRef<int>({1, 2, 3}))` works before because we have: https://github.com/pytorch/pytorch/blob/9f0deb4725dd32f2e9df2828dfbe47e54cbd3b4f/tools/autograd/templates/variable_factories.h#L168-L170
This test is to ensure that it still works after this PR change.

Reply:
It might be worthwhile to test an ArrayRef of another type as well, like `float`.